src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.AnalyzerExecutor.cs

```diff
@@ -94,15 +95,28 @@ public AnalyzerExecutor(DiagnosticIncrementalAnalyzer owner)
                     {
                         var memberDxData = await GetSemanticDiagnosticsAsync(analyzerDriver, stateSet.Analyzer).ConfigureAwait(false);
 
-                        diagnosticData = _owner.UpdateDocumentDiagnostics(existingData, ranges.Ranges, memberDxData.AsImmutableOrEmpty(), root.SyntaxTree, member, memberId);
-                        ValidateMemberDiagnostics(stateSet.Analyzer, document, root, diagnosticData);
+                        // if all the current diagnostics have all their locations in this document,
```

Comment:
I think this method shouldn't be called for analyzer that can't do member update. just make the analyzer never to be in this method.